core/modules/replica/ConfigParserMySQL.h

```diff
@@ -0,0 +1,147 @@
+     * @throws std::runtime_error If a required field has NULL.
+     * @throws std::invalid_argument If the parameter's value didn't pass the validation.
+     */
+    void parse();
+
+private:
+
+    /// Parse general (category/parameter) parameters.
+    void _parseGeneral();
+
+    /**
+     * Parse a collection of workers.
+     *
+     * When parsing optional ports and data folders use default values from
+     * the collection worker defaults. For the optional host names (all but
+     * the name of a host where the replication service 'svc' runs) use
+     * the host name of the 'svc' service.
+     */
+    void _parseWorkers();
+
+    /// Parse a collection of the database families.
+    void _parseDatabaseFamilies();
+
+    /// Parse a collection of the databases.
+    void _parseDatabases();
+
+    /**
+     * Extract a value of the general parameter into the requested type, sanitize the value
+     * if needed, and store it in the transent state.
+     * @throws std::runtime_error If a required field has NULL.
+     * @throws std::invalid_argument If the parameter's value didn't pass the validation.
+     */
+    template <typename T>
+    void _storeGeneralParameter(std::string const& category, std::string const& param) {
+        T value;
+        if (!_row.get("value", value)) {
+            throw std::runtime_error(
+                    _context + " NULL is not allowed, category:'" + category
+                    + "' param: '" + param + "'.");
+        }
+        // Sanitize the input to ensure it matches schema requirements before
+        // pushing the value into the configuration.
+        ConfigurationSchema::validate<T>(category, param, value);
+        _data[category][param] = value;
+    }
+
+    template <typename T>
+    T _parseParam(std::string const& name) {
+        T value;
+        if (_row.get(name, value)) return value;
+        throw std::runtime_error(
+                _context + " the spec field '" + name + "' is not allowed to be NULL");
+    }
+
+    template <typename T>
+    T _parseParam(std::string const& name, T const& defaultValue) {
+        T value;
+        if (_row.get(name, value)) return value;
+        return defaultValue;
+    }
+
+    template <typename T>
+    T _parseParam(std::string const& name, nlohmann::json const& defaults) {
+        T value;
+        if (_row.get(name, value)) return value;
+        return defaults.at(name).get<T>();
+    }
+
+    std::string const _context =  "CONFIG-MYSQL-PARSER  ";
+
+    // Input parameters
+
+    database::mysql::Connection::Ptr const _conn;
+    nlohmann::json& _data;
+    std::map<std::string, WorkerInfo>& _workers;
+    std::map<std::string, DatabaseFamilyInfo>& _databaseFamilies;
+    std::map<std::string, DatabaseInfo>& _databases;
+
+    /// The current row of the MySQL result set is used for extracting
+    /// values of parameters.
+    database::mysql::Row _row;
```

Comment:
It's used by 4 template methods defined in the header of the class in order to extract a value from the current row into a variable of the proposed (by the template argument) type.  I could pass a reference to the row as a parameter of each method. I just decided it would make the code noisier.
```
    template <typename T>
    void _storeGeneralParameter(std::string const& category, std::string const& param)

    template <typename T>
    T _parseParam(std::string const& name)

    template <typename T>
    T _parseParam(std::string const& name, T const& defaultValue)

    template <typename T>
    T _parseParam(std::string const& name, nlohmann::json const& defaults)
```